src/Cake.Common.Tests/Unit/Solution/SolutionParserTests.cs

```diff
@@ -150,5 +151,145 @@ public void Should_Properly_Parse_Projects_With_Absolute_Path()
                 Assert.Single(onlyProjects);
             }
         }
+
+        public sealed class TheParseMethodForSlnx
+        {
+            [Fact]
+            public void Should_Properly_Parse_Projects()
+            {
+                // Given
+                var fixture = new SolutionParserFixture();
+                var slnFilePath = fixture.WithXmlSolutionFile(Resources.SolutionXml_WithProjectsAndFolders);
+                var solutionParser = new SolutionParser(fixture.FileSystem, fixture.Environment);
+
+                // When
+                var result = solutionParser.Parse(slnFilePath);
+
+                // Then
+                Assert.NotNull(result);
+                Assert.NotNull(result.Projects);
+                Assert.Equal(5, result.Projects.Count);
+                var onlyProjects = result.Projects.Where(x => !(x is SolutionFolder)).ToList();
+                Assert.Equal(3, onlyProjects.Count);
```

Comment:
We've started to use Verify for simpler and more complete asserts, currently first in Cake.Core.Tests, but feel this would be a good addition to add it to Cake.Common.Tests, this is optional, I could do the leg work to sort this.

```suggestion
                await Verify(result);
```

Reply:
I just started to have a look at this. I have not used this before, so I am not quite sure what steps I need to take to enable this. But I'll give it a try :)

Reply:
You can see example packages added here
https://github.com/cake-build/cake/blob/develop/src/Cake.Core.Tests/Cake.Core.Tests.csproj#L18-L20

Cake specific settings you set here
https://github.com/cake-build/cake/blob/develop/src/Cake.Core.Tests/VerifyConfig.cs

Believe VerifyDiffPlex.Initialize can only be called  once per solution, so that can be skipped. 

Cake solution doesn't use implicit using, so you need an

using static VerifyXunit.Verifier;



First time you execute tests it'll fail, launch a VSCode diff with snapshot, if correct just copy correct to other side, close and rerun tests.

Reply:
I have migrated the tests for `SlnParser` to use the `Verify` logic you mentioned. Was pretty straight-forward actually. Though I have no idea if there are some details I should look out for. Looks good to me though and tests fail if I change the parsing logic in any way.

Reply:
I just had to rearrange the initialization logic, because:

- When running all tests it would fail with "already initialized"
- But, when running a single test, tests in a test class, in a single test project, it would fail with "needs to be initialized"

So I basically made sure in each `[ModuleInitializer]` to only call `VerifyDiffPlex.Initialize()` when it has not been initialized already (`VerifyDiffPlex.Initialized`).

Let me know if this aligns with the way you would want to see things done.